napari/_qt/dialogs/qt_notification.py

```diff
@@ -436,3 +405,32 @@ def _debug_tb(tb):
         print("Entering debugger. Type 'q' to return to napari.\n")
         pdb.post_mortem(tb)
         print("\nDebugging finished.  Napari active again.")
+
+
+class TracebackDialog(QDialog):
```

Comment:
Is there any benefit to having the new dialog type `TracebackDialog`? It doesn't expose any new public attributes or methods and I think the behavior is the same (i.e. the code here is mostly the same, just swapping `self` with `tbdialog`.

I don't feel strongly here, but just wanted to check I understood the changes.

Reply:
Oh, is it so you can patch it specifically in the test?

Reply:
There are three reasons: 
The Main is to change `_enter_debug_mode` into the method that is better for not leak memory.
The second one you mention: More specific patching. 
And for last it looks better for me.

Reply:
That works.